examples/codes/external_code.ex.py

```diff
@@ -0,0 +1,428 @@
+# ---
+# jupyter:
+#   jupytext:
+#     cell_metadata_filter: tags,-all
+#     notebook_metadata_filter: -jupytext.text_representation.jupytext_version
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#   kernelspec:
+#     display_name: Python 3 (ipykernel)
+#     language: python
+#     name: python3
+# ---
+
+# %% tags=["remove-cell"]
+# bluemira is an integrated inter-disciplinary design tool for future fusion
+# reactors. It incorporates several modules, some of which rely on other
+# codes, to carry out a range of typical conceptual fusion reactor design
+# activities.
+#
+# Copyright (C) 2021-2023 M. Coleman, J. Cook, F. Franza, I.A. Maione, S. McIntosh,
+#                         J. Morris, D. Short
+#
+# bluemira is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# bluemira is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+"""
+External Codes Example
+"""
+# %%
+from dataclasses import asdict, dataclass
+from enum import auto
+from pathlib import Path
+from pprint import pprint
+from typing import Dict, List, Union
+
+# %%
+from ext_code_script import get_filename
+
+from bluemira.base.file import get_bluemira_path
+from bluemira.base.parameter_frame import Parameter, ParameterFrame
+from bluemira.codes.interface import CodesSetup, CodesSolver, CodesTask, CodesTeardown
+from bluemira.codes.interface import RunMode as BaseRunMode
+from bluemira.codes.params import MappedParameterFrame
+from bluemira.codes.utilities import ParameterMapping
```

Comment:
Any reason these imports are in separate cells?

Reply:
Nope :)